File: frontend/types/interview.ts
export type InterviewStatus = "pending" | "in_progress" | "completed" | "failed";

export interface Interview {
  id: string;
  candidateId: string;
  vapiSessionId?: string;
  status: InterviewStatus;
  startedAt?: string;
  completedAt?: string;
  durationSeconds?: number;
  audioUrl?: string;
  fullTranscript?: unknown;
  questionsAsked?: unknown;
  createdAt: string;
  updatedAt: string;
}

File: backend/app/models/interview.py
from pydantic import BaseModel
from typing import Optional


class InterviewCreate(BaseModel):
    candidate_id: str


class Interview(BaseModel):
    id: str
    candidate_id: str
    vapi_session_id: Optional[str] = None
    status: str = "pending"

    class Config:
        from_attributes = True
